core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -3063,24 +3071,53 @@ private CompiledQuery sqlBackup(SqlExecutionContext executionContext) throws Sql
         }
 
         private CompiledQuery sqlDatabaseBackup(SqlExecutionContext executionContext) {
-            tableTokens.clear();
-            setupBackupRenamePath();
-            cdDbRenamePath();
-            engine.getTableTokens(tableTokens, false);
+            mkBackupDstRoot();
+            mkBackupDstDir(configuration.getDbDirectory(), "could not create backup [db dir=");
+
+            // backup tables
+            engine.getTableTokens(tableTokenBucket, false);
             executionContext.getSecurityContext().authorizeTableBackup(tableTokens);
-            for (int i = 0, n = tableTokens.size(); i < n; i++) {
-                backupTable(tableTokens.get(i));
+            for (int i = 0, n = tableTokenBucket.size(); i < n; i++) {
+                backupTable(tableTokenBucket.get(i));
             }
-            backupTabIndexFile();
-            cdConfRenamePath();
-            ff.iterateDir(srcPath.of(configuration.getConfRoot()).$(), confFilesBackupOnFind);
+
+            srcPath.of(configuration.getRoot()).$();
+            int srcLen = srcPath.length();
+
+            // backup table registry file (tables.d.<last>)
+            int version = TableNameRegistryFileStore.findLastTablesFileVersion(ff, srcPath, sink);
+            srcPath.trimTo(srcLen).concat(WalUtils.TABLE_REGISTRY_NAME_FILE).put('.').put(version).$();
+            dstPath.trimTo(dstCurrDirLen).concat(WalUtils.TABLE_REGISTRY_NAME_FILE).put(".0").$(); // reset to 0
+            LOG.info().$("backup copying file [from=").utf8(srcPath).$(", to=").utf8(dstPath).I$();
+            if (ff.copy(srcPath, dstPath) < 0) {
+                throw CairoException.critical(ff.errno())
+                        .put("cannot backup table registry file [from=").put(srcPath)
+                        .put("m to=").put(dstPath)
```

Comment:
```suggestion
                        .put(" to=").put(dstPath)
```

Reply:
addressed in 8fcbde869d126ea12ec4e1ee5981636a2a19dc61